lldb/include/lldb/API/SBSaveCoreOptions.h

```diff
@@ -80,6 +80,17 @@ class LLDB_API SBSaveCoreOptions {
   /// \return True if the thread was removed, false if it was not in the list.
   bool RemoveThread(lldb::SBThread thread);
 
+  /// Add a memory region to save in the core file.
+  ///
+  /// \param region The memory region to save.
+  /// \returns An empty SBError upon success, or an error if the region is
+  /// invalid.
+  /// \note Ranges that overlapped with be unioned into a single region this
```

Comment:
`with be` -> `will be`

Reply:
"Ranges that overlap will be unioned into a single region, this also supercedes stack minification."

Some other things we could do with this patch:
- if a region crosses a permission boundary, then break it up so we can safe off memory with the right permissions. So if we have [0x1000-0x2000) with "rw" permissions and [0x2000-0x3000) with "r" permissions, and the user asks for  [0x1000-0x3000) to be saved, we should probably break it up and save them separately.
- Allow the user to ask for a huge range and break it up into any regions that fit, like ask for `[0x100000000-0xFFFFFFFFFFFFFFFF)` and we would save all regions in that range, separated out by permission boundaries